core/opentype-parser.lua

```diff
@@ -188,15 +188,262 @@ local function parseSvg(str)
+  end
+  local mathConstants = vstruct.read(mathConstantFormat, fd)
+  for _,v in pairs(mathConstants) do
+    if v and type(v) == "table" then
+      fetchMathValueRecord(v, offset, fd)
+    end
+  end
+  return mathConstants
+end
+
+local parseMathKern = function(offset, fd)
+  local heightCount	= vstruct.readvals(">@"..offset.." u2", fd)
+  local mathKern = vstruct.read("> correctionHeight:{ "..heightCount.."*{ &MathValueRecord } } kernValues:{ "..(heightCount+1).."*{ &MathValueRecord } }", fd)
+  for i = 1, #(mathKern.correctionHeight) do
+    fetchMathValueRecord(mathKern.correctionHeight[i], offset, fd)
+  end
+  for i = 1, #(mathKern.kernValues) do
+    fetchMathValueRecord(mathKern.kernValues[i], offset, fd)
+  end
+  return mathKern
+end
+
+local parsePerGlyphTable = function(offset, type, fd)
+  local coverageOffset = vstruct.readvals(">@"..offset.." u2", fd)
+  local coverageTable = parseCoverage(offset + coverageOffset, fd)
+  local count = vstruct.readvals(">@"..(offset+2).." u2", fd)
+  if count ~= #(coverageTable) then
+    SU.error("Coverage table corrupted")
+  end
+  local table = vstruct.read("> "..count.."*{ "..type.." }", fd)
+  local result = {}
+  for i = 1, count do
+    if type == "&MathValueRecord" then
+      fetchMathValueRecord(table[i], offset, fd)
+      result[coverageTable[i]] = table[i]
+    elseif type == "&MathKernInfoRecord" then
+      result[coverageTable[i]] = {
+        topRightMathKern = table[i].topRightMathKernOffset ~= 0 and parseMathKern(offset + table[i].topRightMathKernOffset, fd) or nil,
+        topLeftMathKern = table[i].topLeftMathKernOffset ~= 0 and parseMathKern(offset + table[i].topLeftMathKernOffset, fd) or nil,
+        bottomRightMathKern =  table[i].bottomRightMathKernOffset ~= 0 and parseMathKern(offset + table[i].bottomRightMathKernOffset, fd) or nil,
+        bottomLeftMathKern = table[i].bottomLeftMathKernOffset ~= 0 and parseMathKern(offset + table[i].bottomLeftMathKernOffset, fd) or nil
+      }
+    else
+      result[coverageTable[i]] = table[i]
+    end
+  end
+  return result
+end
+
+-- TODO: Figure out why this function existed and fix it or remove it
+-- local parseGlyphAssembly = function(offset, fd)
+local parseGlyphAssembly = function(_, _)
+  -- local assembly = vstruct.read(">@"..offset.." italicsCorrection:{ &MathValueRecord } partCount:u2", fd)
+  -- fetchMathValueRecord(assembly.italicsCorrection, offset, fd)
+  -- assembly.partRecords = vstruct.read("> "..assembly.partCount.."*{ &GlyphPartRecord }", fd)
+  -- assembly.partCount = nil
+  return nil
+end
+
+local parseMathGlyphConstruction = function(offset, fd)
+  local construction = vstruct.read(">@"..offset.." glyphAssemblyOffset:u2 variantCount:u2", fd)
+  local mathGlyphVariantRecord = vstruct.read("> "..construction.variantCount.."*{ &MathGlyphVariantRecord }", fd)
+  return {
+    glyphAssembly = construction.glyphAssemblyOffset ~= 0 and parseGlyphAssembly(offset + construction.glyphAssemblyOffset, fd) or nil,
```

Comment:
The `parseGlyphAssembly()` function never returns anything but `nil` no matter what you pass it, and it doesn't even create any side effects. I replaced it with a literal nil to demonstrate (and get rid of the lint errors) but I don't understand what it was intended to do so I can't fix it properly.

Reply:
I reverted some of my lint fixes because they broke the code. They broke the code because some of the functions are relying on side effects. Those will need to get refactored before this can be merged.